source/extensions/filters/http/ext_authz/config.cc

```diff
@@ -57,15 +57,15 @@ Http::FilterFactoryCb ExtAuthzFilterConfig::createFilterFactoryFromProtoTyped(
     };
   } else {
     // Envoy gRPC client.
-
-    Grpc::RawAsyncClientSharedPtr raw_client =
-        context.clusterManager().grpcAsyncClientManager().getOrCreateRawAsyncClient(
-            proto_config.grpc_service(), context.scope(), true, Grpc::CacheOption::AlwaysCache);
+    
     const uint32_t timeout_ms =
         PROTOBUF_GET_MS_OR_DEFAULT(proto_config.grpc_service(), timeout, DefaultTimeout);
-    callback = [raw_client, filter_config, timeout_ms,
+    callback = [grpc_service = proto_config.grpc_service(), &context, filter_config, timeout_ms,
```

Comment:
It would be helpful while reading this to add more comments or explicitly typed temps to understand better the lifetimes of the variables involved.

I have confidence in the lifetime of filter_config because I can tell from local inspection that it is a shared_ptr. But I can't tell what grpc_service is just by reading this code.

Reply:
the grpc_service is capturing a copy here.

Reply:
I know; I'm not complaining about the code; I was just asking more comments be added to the code for clarity.

For example if grpc_service was a pointer, capturing a copy wouldn't help with lifetime.

Reply:
Comments added at the beginning of the function.